shared/rakefiles/secrets.rb

```diff
@@ -119,12 +123,28 @@ def self.push_secrets(secrets, encryption_key)
     gs_bucket = "#{ENV['TF_VAR_project_id']}-#{encryption_key}-secrets"
     encoded_secrets = Base64.encode64(secrets.to_json).delete!("\n")
 
+    puts "[secret-mgmt] Retrieving primary key version for key '#{encryption_key}'..."
+    encryption_key_version = %x{
+      curl -s \
+      -H \"Authorization:Bearer $(gcloud auth print-access-token)\" \
+      -H \"Content-Type:application/json\" \
+      -X GET \"#{Secrets::GOOGLE_KMS_API}/v1/projects/#{ENV['TF_VAR_project_id']}/locations/#{Secrets::KMS_LOCATION}/keyRings/#{Secrets::KMS_KEYRING}/cryptoKeys/#{encryption_key}\"
+    }
+
+    begin
+      encryption_key_version = JSON.parse(encryption_key_version)
+      encryption_key_version = encryption_key_version['primary']['name'].match(/\/([0-9]+)$/)[1]
+    rescue
+      puts "ERROR: Unable to get primary encryption key version for key '#{encryption_key}', terminating!"
```

Comment:
If it's safe to do so, it might be helpful to emit the JSON blob as part of this failure to assist debugging.

Reply:
Done.